projects/transactions-journey/src/lib/views/transactions-view/transactions-view.component.spec.ts

```diff
@@ -0,0 +1,145 @@
+  @Input() transaction: Transaction[] | undefined;
+}
+
+@Component({
+  selector: 'bb-text-filter',
+  template: '<div></div>'
+})
+class FakeTextFilter {
+  @Output() textChange = new EventEmitter<string>();
+}
+
+const transactions = [{
+    categoryCode: '#12a580',
+    dates: {
+      valueDate: 1600493600000
+    },
+    transaction: {
+      amountCurrency: {
+        amount: 5000,
+        currencyCode: 'EUR'
+      },
+      type: 'Salaries',
+      creditDebitIndicator: 'CRDT'
+    },
+    merchant: {
+      name: 'Backbase',
+      accountNumber: 'SI64397745065188826'
+    }
+  }, {
+    categoryCode: '#12a580',
+    dates: {
+      valueDate: 1600387200000
+    },
+    transaction: {
+      amountCurrency: {
+        amount: 82.02,
+        currencyCode: 'EUR'
+      },
+      type: 'Card Payment',
+      creditDebitIndicator: 'DBIT'
+    },
+    merchant: {
+      name: 'The Tea Lounge',
+      accountNumber: 'SI64397745065188826'
+    }
+  }, {
+    categoryCode: '#d51271',
+    dates: {
+      valueDate: 1600473600000
+    },
+    transaction: {
+      amountCurrency: {
+        amount: 84.64,
+        currencyCode: 'EUR'
+      },
+      type: 'Card Payment',
+      creditDebitIndicator: 'DBIT'
+    },
+    merchant: {
+      name: 'Texaco',
+      accountNumber: 'SI64397745065188826'
+    }
+  }];
+
+  const additionalTransaction = {
+    categoryCode: '#fbbb1b',
+    dates: {
+      valueDate: 1599868800000
+    },
+    transaction: {
+      amountCurrency: {
+        amount: 142.95,
+        currencyCode: 'EUR'
+      },
+      type: 'Online Transfer',
+      creditDebitIndicator: 'DBIT'
+    },
+    merchant: {
+      name: 'Southern Electric Company',
+      accountNumber: 'SI64397745065188826'
+    }
+  };
+describe('TransactionsViewComponent', () => {
+  let fixture: ComponentFixture<TransactionsViewComponent>;
+  let component: TransactionsViewComponent;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [CommonModule, RouterTestingModule],
+      declarations: [FakeTransactionsItem, FakeTextFilter, FilterTransactionsPipe, TransactionsViewComponent],
+      providers: [
+        {
+          provide: TransactionsHttpService,
+          useValue: {
+            transactions$: of(transactions)
+          },
+        },
+        {
+          provide: TRANSACTIONS_JOURNEY_COMMUNICATION_SERIVCE,
+          useValue: {
+            latestTransaction$: of(additionalTransaction),
+          }
+        }
+      ],
+    });
+
+    fixture = TestBed.createComponent(TransactionsViewComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should load all the transactions from both communciation service and http service', () => {
+    const result = fixture.debugElement.queryAll(By.css('.bb-list__item'));
+
+    expect(result.length).toBe(4);
+  });
+
+  it('should filter transactions based on the search', () => {
+    component.filter = 'Tea';
```

Comment:
We can be closer to the actual flow of the working component if this form will be used:
`fixture.debugElement.query(By.css('bb-text-filter')).triggerEventHandler('textChange', 'Tea');`

Reply:
done